lib/pwork/async.rb

```diff
@@ -0,0 +1,50 @@
+require_relative 'async/task'
+require_relative 'async/exceptions'
+require_relative 'async/manager'
+
+module PWork
+  module Async
+    def async(options = nil, &block)
+      if block_given?
+        PWork::Async.add_task(options, &block)
+      else
+        raise PWork::Async::Exceptions::InvalidOptionsError.new(
+          'Unknown async option.'
+        ) unless options == :wait
+        PWork::Async.wait_for_tasks
+      end
+    end
+
+    def self.manager
+      @manager ||= PWork::Async::Manager.new
+    end
+
+    def self.add_task(options, &block)
+      manager.start unless manager.running
+
+      options = {} if options == nil
+      task = PWork::Async::Task.new.tap do |e|
+        e.block = block
+      end
+
+      unless options[:wait] == false
+        tasks << task
+      end
+
+      manager.add_task(task)
+
+      task.id
+    end
+
+    def self.tasks
+      Thread.current[:pwork_async_tasks] ||= []
+    end
+
+    def self.wait_for_tasks
+      until tasks.detect { |t| t.state == :pending || t.state == :active }.nil?
+        sleep(0.01)
```

Comment:
Would this be too fast? We were using `0.02` in gateway system pooling and it proved to consume a lot of CPU.